tests/pipeline/controlflowanalysis/test_readability_based_refinement.py

```diff
@@ -894,10 +898,217 @@ def test_skip_for_loop_recovery_if_continue_in_while(self):
         )
 
         WhileLoopReplacer(ast, _generate_options()).run()
-        assert not any(isinstance(loop_node, ForLoopNode) for loop_node in list(ast.get_loop_nodes_post_order()))
+        assert all(isinstance(loop_node, ForLoopNode) for loop_node in list(ast.get_loop_nodes_post_order()))
+
+        condition_nodes = list(ast.get_condition_nodes_post_order())
+        last_definition = condition_nodes[0].true_branch_child.instructions[
+            _get_last_definition_index_of(condition_nodes[0].true_branch_child, Variable("a"))
+        ]
+        assert last_definition == Assignment(
+            Variable("a"),
+            BinaryOperation(OperationType.minus, [BinaryOperation(OperationType.plus, [Variable("a"), Constant(2)]), Constant(1)]),
+        )
 
-    def test_skip_for_loop_recovery_if_continue_in_nested_while(self):
+    def test_for_loop_recovery_if_continue_in_while_2(self):
         """
+        Test for loop recovery if a continue occurs in a while loop and the last definition is a constant.
+
+        a = 0
+        while(a < 10) {
+            if(a == 2) {
+                a = 4
+                continue
+            }
+            a = a + 1
+        }
```

Comment:
We skip such a case because the increment-function isn't a simple binary operation. I added a test for this in [d4d5dd7](https://github.com/fkie-cad/dewolf/pull/326/commits/d4d5dd71349e498350a56973e82ba41c6da8b388)